src/Generator/Changelog/Formats/Json.php

```diff
@@ -0,0 +1,202 @@
+    {
+        $entries = [];
+        foreach ($changesets as $representation => $change_types) {
+            foreach ($change_types as $change_type => $hashes) {
+                foreach ($hashes as $hash => $changes) {
+                    if ($definition === 'added' || $definition === 'removed') {
+                        $entry = $this->getEntryForAddedOrRemovedChange($definition, $change_type, $changes);
+                    } else {
+                        $entry = $this->getEntryForChangedItem($definition, $change_type, $changes);
+                    }
+
+                    // Reduce some unnecessary nesting of changeset strings.
+                    if (is_array($entry) && count($entry) === 1) {
+                        $entry = array_shift($entry);
+                    }
+
+                    $entries[] = $entry;
+                }
+            }
+        }
+
+        return $entries;
+    }
+
+    /**
+     * Parse resource changesets.
+     *
+     * @param string $definition
+     * @param array $changesets
+     * @return array
+     */
+    private function parseResourceChangesets($definition, array $changesets = [])
+    {
+        $entries = [];
+        foreach ($changesets as $group => $data) {
+            $group_entry = [
+                $this->renderText('The following {group} resources have ' . $definition . ':', [
+                    'group' => $group
+                ]),
+                [] // Group-related entries will be nested here.
+            ];
+
+            foreach ($data as /*$uri => */$change_types) {
+                foreach ($change_types as $change_type => $hashes) {
+                    foreach ($hashes as /*$hash => */$changes) {
+                        if ($definition === 'added' || $definition === 'removed') {
+                            $entry = $this->getEntryForAddedOrRemovedChange($definition, $change_type, $changes);
+                        } else {
+                            $entry = $this->getEntryForChangedItem($definition, $change_type, $changes);
+                        }
+
+                        // Reduce some unnecessary nesting of changeset strings.
+                        if (is_array($entry) && count($entry) === 1) {
+                            $entry = array_shift($entry);
+                        }
+
+                        $group_entry[1][] = $entry;
+                    }
+                }
+            }
+
+            $entries[] = $group_entry;
+        }
+
+        return $entries;
+    }
+
+    /**
+     * Get a changelog entry for a changeset that was added into the API.
+     *
+     * @param string $definition
+     * @param string $change_type
+     * @param array $changes
+     * @return string|array
+     * @throws \Exception If an unsupported definition + change type was supplied.
+     */
+    private function getEntryForAddedOrRemovedChange($definition, $change_type, array $changes)
```

Comment:
Renaming to `getAddedOrRemovedChangesetFactory` and `getChangedChangesetFactory`.